Automattic/wp-calypso
# Inline Help/Help Center: use the same useHelpSearchQuery hook

The default search results for the inline help are not currently loading: 

<img width="513" alt="Screenshot 2023-06-08 at 13 18 01" src="https://github.com/Automattic/wp-calypso/assets/844866/49b13a07-0b5d-4679-bab1-5161b2376b9f">

Reported in p9Jlb4-7c4-p2#comment-8216


## Proposed Changes

* This updates calypso useHelpSearchQuery hook to use the new endpoint (see https://github.com/Automattic/wp-calypso/pull/76734/ )
* It then removes the custom hook in the Help Center, and uses the calypso hook instead.
* Because the `/help` screen search results need data that the new endpoint does not provide, this also brings back the old hook as `-v1` and uses it for this screen.

## Testing Instructions

* In calypso home, make sure you see the default search results in the help widget:

<img width="500" alt="Screenshot 2023-06-08 at 13 19 32" src="https://github.com/Automattic/wp-calypso/assets/844866/dc59f226-59dc-4a07-ab10-c72205ffd49d">

* In `/help` search results should include both Documentation and Forum links

* The Help Center default results (and search results) should stay the same.

## Pre-merge Checklist
